os/src/Path.scala

```diff
@@ -5,29 +5,52 @@ import java.nio.file.Paths
 
 import collection.JavaConverters._
 import scala.language.implicitConversions
-import java.nio.file
+import acyclic.skipped //needed for cross-version defined macros
 
 trait PathChunk {
   def segments: Seq[String]
   def ups: Int
 }
-object PathChunk {
+trait StringPathChunkConversion {
+
+  implicit def stringToPathChunk(s: String): PathChunk =
+    new PathChunk.StringPathChunkInternal(s)
+}
+
+object PathChunk extends PathChunkMacros {
+  def segmentsFromString(s: String): Array[String] = {
```

Comment:
@lihaoyi this is public only to make tests possible, however if you are fine with deleting tests for it, we can reduce accessibility